src/main/java/co/omise/models/Source.java

```diff
@@ -0,0 +1,111 @@
+package co.omise.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Source extends Model {
+
+    private String type;
+    private String flow;
```

Comment:
Can we use the Type System to improve the overall library experience?

Reply:
Yes, we can I'll look into na krub. 😄

Reply:
@pitiphong-p There's no 1st-class string enum in Java 😂

Reply:
How's about just an ordinary enum?

Reply:
How about this?
```java
public enum SourceType {
    InternetBankingScb,
    Alipay,
    BillPaymentTesco,
    VirtualAccountSinarmas,
    ...
}
```

Reply:
My initial thought is something like

```java
public enum SourceType {
    InternetBanking,
    Alipay,
    BillPayment,
    VirtualAccount,
    ...
}
```

and then we may have another property for those subtypes

Reply:
Rite, ordinary enum can work. I thought we're going to need a `toString()` but maybe just `@JsonProperty("internet_banking")` should work.

Reply:
I have to override `toString()` too.

Reply:
Would you like to change currency too.

Reply:
Yeah, the problem with this approach is how to design a beautiful API for those subtypes Orz